llama_index/ingestion/pipeline.py

```diff
@@ -195,8 +251,51 @@ def run(
         if self.reader is not None:
             input_nodes += self.reader.read()
 
+        return input_nodes
+
+    def run(
+        self,
+        show_progress: bool = False,
+        documents: Optional[List[Document]] = None,
+        nodes: Optional[List[BaseNode]] = None,
+        cache_collection: Optional[str] = None,
+        in_place: bool = True,
+        **kwargs: Any,
+    ) -> Sequence[BaseNode]:
+        input_nodes = self._prepare_inputs(documents, nodes)
```

Comment:
after we get input_nodes, we can do in bulk update. Set a batch size and split input_nodes for batches for updates?